pkg/objectio/block.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package objectio
+
+import (
+	"bytes"
+	"encoding/binary"
+	"github.com/matrixorigin/matrixone/pkg/container/batch"
+)
+
+// Block is the organizational structure of a batch in objectio
+// Write one batch at a time, and batch and block correspond one-to-one
+type Block struct {
+	// fd is the handle of the block
+	fd     int
+	header *BlockHeader
+	// columns is the vector in the batch
+	columns []ColumnObject
+	// data is the batch to be written
+	data   *batch.Batch
+	object *Object
+	extent Extent
+}
+
+func NewBlock(batch *batch.Batch, object *Object) BlockObject {
+	header := &BlockHeader{
+		columnCount: uint16(len(batch.Vecs)),
+	}
+	block := &Block{
+		header:  header,
+		data:    batch,
+		object:  object,
+		columns: make([]ColumnObject, len(batch.Vecs)),
+	}
+	for i := range block.columns {
+		block.columns[i] = NewColumnBlock(uint16(i), block)
+	}
+	return block
+}
+
+func (b *Block) GetExtent() Extent {
+	return b.extent
+}
+
+func (b *Block) GetColumn(idx uint16) (ColumnObject, error) {
+	return b.columns[idx], nil
+}
+
+func (b *Block) GetRows() (uint32, error) {
+	panic(any("implement me"))
+}
+
+func (b *Block) GetMeta() *BlockMeta {
+	return &BlockMeta{
+		header: b.header,
+	}
+}
+
+func (b *Block) MarshalMeta() ([]byte, error) {
+	var (
+		err    error
+		buffer bytes.Buffer
+	)
+	// write header
```

Comment:
Is there a reason to use BigEndian?   Most likely our code runs on little endian machines.